modules/publicize/rest-api.php

```diff
@@ -0,0 +1,330 @@
+	 * send them as JSON encoded data.
+	 *
+	 * @see Publicize::get_available_service_data()
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return string JSON encoded connection services data.
+	 */
+	public function rest_get_publicize_available_services() {
+		/**
+		 * We need this because Publicize::get_available_service_data() uses `Jetpack_Keyring_Service_Helper`
+		 * and `Jetpack_Keyring_Service_Helper` relies on `menu_page_url()`.
+		 *
+		 * We also need add_submenu_page(), as the URLs for connecting each service
+		 * rely on the `sharing` menu subpage being present.
+		 */
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+
+		// The `sharing` submenu page must exist for service connect URLs to be correct.
+		add_submenu_page( 'options-general.php', '', '', 'manage_options', 'sharing', '__return_empty_string' );
+
+		return $this->publicize->get_available_service_data();
+	}
+
+	/**
+	 * Add rest field to 'post' for Publicize support
+	 *
+	 * Sets up 'publicize' schema to submit publicize sharing title
+	 * and individual connection sharing enables/disables. This schema
+	 * is registered with the 'post' endpoint REST endpoint so publicize
+	 * options can be saved when a post is published.
+	 *
+	 * @since 5.9.1
+	 */
+	public function add_publicize_rest_fields() {
+		// Schema for wpas.submit[] field.
+		$publicize_submit_schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => esc_html__( 'Publicize data for publishing post', 'jetpack' ),
+			'type'       => 'object',
+			'properties' => array(
+				'connections' => array(
+					'description' => esc_html__( 'List of connections to be shared to (or not).', 'jetpack' ),
+					'type'        => 'array',
+					'items'       => array(
+						'type'       => 'object',
+						'properties' => array(
+							'unique_id'    => array(
+								'description' => esc_html__( 'Unique identifier string for a connection', 'jetpack' ),
+								'type'        => 'string',
+							),
+							'should_share' => array(
+								'description' => esc_html__( 'Whether or not connection should be shared to.', 'jetpack' ),
+								'type'        => 'boolean',
+							),
+
+						),
+					),
+				),
+				'title'       => array(
+					'description' => esc_html__( 'Optional title to share post with.', 'jetpack' ),
+					'type'        => 'string',
+				),
+			),
+		);
+
+		// Registering the publicize field with post endpoint.
+		register_rest_field(
+			'post',
+			'publicize',
+			array(
+				'get_callback'    => null,
+				'update_callback' => null, // Data read/processed before publishing post by 'rest_pre_insert_post' filter.
+				'schema'          => $publicize_submit_schema,
+			)
+		);
+
+		/**
+		 * REST endpoint to get connection list data for current user.
+		 *
+		 * @see Publicize::get_filtered_connection_data()
+		 *
+		 * @since 5.9.1
+		 */
+		register_rest_route( 'publicize/', '/connections', array(
```

Comment:
Yep, right now it's only copy-pasta from #10372's `class-jetpack-publicize-gutenberg.php`. I'm planning on moving the endpoint registrations into `class.core-rest-api-endpoints.php` since that seems to be where all the other endpoints are.

However, turns out that they're all in the `jetpack/v4` namespace, so I was inclined to stick with that. I'm not sure what the overall plans are for these...